### ISSUE TITLE
local variable between #ifdef and #endif can't go to definition if macro isn't defined.

### ISSUE THREAD

**Issue by Lennon925** [microsoft/vscode-cpptools#3108]
**Type: LanguageService**

**Describe the bug**
- OS and Version:
- VS Code Version: 1.30.2
- C/C++ Extension Version: 0.21.0
- Other extensions you installed (and if the issue persists after disabling them):
- A clear and concise description of what the bug is.

codes as blow:

#define DEBUG_CHECK_RANGE 1;

void test()
{
    int count = 0;
#ifdef DEBUG_CHECK_RANGE0   // it's not defined
    int local = 0;
    local++;         // can't go to definition
    count++;       // can't go to definition
#endif
    count++;      //it's ok
}

**Comment by bobbrow** [comment]
The code inside the inactive preprocessor region is not compiled, so goto defintion cannot work for that case.  This is currently the expected behavior.

**Comment by baoyuncheng** [comment]
I don't think that is reasonable, if there are hundreds macro defined inside makefile or other files which already existed in my work project... As i just want to goto the definitions defined in "#ifdef and #endif code", vscode could not support that.
depressed... is there anyway to solve it?
Tips: My source insight works well for that.

**Comment by sean-mcmanus** [comment]
VS itself has a semi-workaround via Find All References, see https://github.com/microsoft/vscode-cpptools/issues/5398 .

I've filed a feature request on VS for Go to Def in inactive code: https://developercommunity.visualstudio.com/idea/1015351/enable-c-go-to-def-on-inactive-code.html

**Comment by wy-source** [comment]
I also need this function. I can make my work more convenient.  Now, I have to use Global Search instead which consumes more time.

**Comment by bobbrow** [comment]
Thank you for the feature idea. At this time we do not plan to move forward with this, so I will close it for now.
